src/connector/src/source/filesystem/file_common.rs

```diff
@@ -79,7 +79,8 @@ impl FsSplit {
 pub struct OpendalFsSplit<Src: OpendalSource> {
     pub name: String,
     pub offset: usize,
-    pub size: usize,
+    // For Parquet encoding, the `end_offset` represents the number of rows, while for other encodings, the `end_offset` denotes the file size.
+    pub end_offset: usize,
```

Comment:
This will break compatibility. You can try `#[serde(alias = "size")]` and verify whether it can parse the state table row written prior to this PR.

Reply:
To be safe, let's keep the original naming.